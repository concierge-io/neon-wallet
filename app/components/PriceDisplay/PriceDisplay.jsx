// @flow
import React from 'react'
import { formatFiat } from '../../core/formatters'
import styles from './PriceDisplay.scss'
import classNames from 'classnames'

type Props = {
  neoPrice: number,
  gasPrice: number
}

const PriceDisplay = ({ neoPrice, gasPrice }: Props) => {
  let neoDisplayPrice = '-'
  let gasDisplayPrice = '-'

  if (neoPrice) {
    neoDisplayPrice = formatFiat(neoPrice)
  }

  if (gasPrice) {
    gasDisplayPrice = formatFiat(gasPrice)
  }

  return (
    <div className={styles.container}>
      <span id='neoPrice' className={styles.neoPriceDisplay}>
        <span className={styles.label}>NEO</span>
        <span className={classNames('price', styles.price)}>${neoDisplayPrice}</span>
      </span>
      <span id='gasPrice'>
        <span className={styles.label}>GAS</span>
        <span className={classNames('price', styles.price)}>${gasDisplayPrice}</span>
      </span>
    </div>
  )
}

export default PriceDisplay
