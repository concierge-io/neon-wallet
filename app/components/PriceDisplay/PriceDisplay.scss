@import '../../styles/variables';

.container {
    position: absolute;
    font-size: .8em;
    opacity: 0.6;
    right: 463px;
    top: 13px;
}

.label {
    opacity: .6;
    margin-right: 4px;
}

.price {
    color: $thin-text-color;
    opacity: .8;
    cursor: pointer;
}

.neoPriceDisplay {
    padding-right: 38px
}