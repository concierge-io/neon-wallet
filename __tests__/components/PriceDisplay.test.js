import React from 'react'
import { mount } from 'enzyme'
import PriceDisplay from '../../app/components/PriceDisplay'

describe('PriceDisplay', () => {
  test('renders and shows prices', (done) => {
    // Full mount (not shallow) so the snapshot will contain the prices in the rendered html.
    const wrapper = mount(<PriceDisplay neoPrice={25.48} gasPrice={18.10} />)
    expect(wrapper).toMatchSnapshot()
    done()
  })
})
