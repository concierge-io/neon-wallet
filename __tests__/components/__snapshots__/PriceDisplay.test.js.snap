// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDisplay renders and shows prices 1`] = `
<PriceDisplay
  gasPrice={18.1}
  neoPrice={25.48}
>
  <div
    className="container"
  >
    <span
      className="neoPriceDisplay"
      id="neoPrice"
    >
      <span
        className="label"
      >
        NEO
      </span>
      <span
        className="price price"
      >
        $
        25.48
      </span>
    </span>
    <span
      id="gasPrice"
    >
      <span
        className="label"
      >
        GAS
      </span>
      <span
        className="price price"
      >
        $
        18.10
      </span>
    </span>
  </div>
</PriceDisplay>
`;
